### DISCUSSION TITLE
Contibuting Custom Nodes (like nodeTypes in reactflow)

### DISCUSSION THREAD

**Discussion by PapaTemporal** [open-source-labs/Svelvet#154] [General]
Is there a trick to add a prop to a `<Svelvet />` component? 

I have added the `export let customNodes: customNodesType;` to Containers/Svelvet/index.svelte but when I package, it doesn't seem to add it to the typescript definition. This results in an `<Svelvet> was created with unknown prop 'customNodes'` message in the browser when I create an <Svelvet /> element with the new customNodes prop.

What's odd to me is that the index.d.ts says:

`export default Svelvet;`
`import Svelvet from "svelvet";`

which imports Svelvet from the npm svelvet instead of its own Containers/Svelvet. Maybe this could be the problem?

Any ideas?

**Comment by rohanrajpal** [comment]
I think @bryanmylee might be able to help you out with this. He has built a [wonderful library](https://github.com/bryanmylee/svelte-headless-table) which [allows sending props](https://svelte-headless-table.bryanmylee.com/docs/api/create-render#usage) to custom Svelte components. Which itself is based on [svelte-render.](https://github.com/bryanmylee/svelte-render)

**Reply by PapaTemporal** [reply]
Thanks! Actually, after I created this I wanted to remove it, but there is no "delete discussion" :). Or maybe I just missed it somewhere.
